File: Animation-OC/Animation-OC/EasingViewController.m

//
//  EasingViewController.m
//  Animation-OC
//
//  Created by chenyufeng on 15/12/31.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

#import "EasingViewController.h"

@interface EasingViewController ()

@property (weak, nonatomic) IBOutlet UIView *greenSquare;

@property (weak, nonatomic) IBOutlet UIView *redSquare;


@end

@implementation EasingViewController

- (void)viewDidLoad {
    [super viewDidLoad];

  
}

- (void)viewWillAppear:(BOOL)animated{

  [super viewWillAppear:animated];

  //主要是设置options这个参数；
  [UIView animateWithDuration:2 delay:0 options:UIViewAnimationOptionCurveEaseIn animations:^{

    self.greenSquare.frame = CGRectMake(250, self.greenSquare.frame.origin.y, self.greenSquare.frame.size.width, self.greenSquare.frame.size.height);

  } completion:nil];

  [UIView animateWithDuration:2 delay:0 options:UIViewAnimationOptionCurveEaseOut animations:^{

    self.redSquare.frame = CGRectMake(250, self.redSquare.frame.origin.y, self.redSquare.frame.size.width, self.redSquare.frame.size.height);

  } completion:nil];
  
}

@end

File: PositionAnimation01/PositionAnimation01/EasingViewController.swift
//
//  EasingViewController.swift
//  PositionAnimation01
//
//  Created by chenyufeng on 15/11/9.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

import UIKit

class EasingViewController: UIViewController {

  
  @IBOutlet weak var blueSquare: UIView!
  @IBOutlet weak var redSquare: UIView!
  @IBOutlet weak var greenSquare: UIView!
  
  @IBOutlet weak var yellowSquare: UIView!
  
  
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }


  override func viewDidAppear(animated: Bool) {
    super.viewDidAppear(animated);
    
    UIView.animateWithDuration(1) { () -> Void in
      
      self.blueSquare.center.x = self.view.bounds.width - self.blueSquare.center.x
    }
    
    UIView.animateWithDuration(1, delay: 0, options: UIViewAnimationOptions.CurveEaseIn, animations: { () -> Void in
      self.redSquare.center.x = self.view.bounds.width - self.redSquare.center.x
      }, completion: nil)
    
    UIView.animateWithDuration(1, delay: 0, options: UIViewAnimationOptions.CurveEaseOut, animations: { () -> Void in
      self.greenSquare.center.x = self.view.bounds.width - self.greenSquare.center.x
      }, completion: nil)
    
    UIView.animateWithDuration(1, delay: 0, options: UIViewAnimationOptions.CurveEaseInOut, animations: { () -> Void in
      self.yellowSquare.center.x = self.view.bounds.width - self.yellowSquare.center.x
      }, completion: nil)
    
    
    
  }
  


}
